source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst

```diff
@@ -65,16 +66,22 @@ Seedlist Discovery
 ------------------
 
 In this preprocessing step, the driver will query the DNS server for SRV
-records on ``{hostname}.{domainname}``, prefixed with ``_mongodb._tcp.``:
-``_mongodb._tcp.{hostname}.{domainname}``. This DNS query is expected to
-respond with one or more SRV records. From the DNS result, the driver now MUST
-behave the same as if an ``mongodb://`` URI was provided with all the host
+records on ``{hostname}.{domainname}``.
+
+If ``mongodb+srv`` is used, ``{hostname}.{domainname}`` MUST be prefixed with
+``_mongodb._tcp.``: ``_mongodb._tcp.{hostname}.{domainname}``. If ``mongodb+rawsrv``
+is used, the user has supplied their own service name and protocol in the URI, and
+``_mongodb._tcp`` SHALL NOT be prepended. If the ``mongodb+rawsrv`` scheme is used
+and no service name has been provided in the URI, the driver MUST raise a parse error.
```

Comment:
I read the old comments in [DRIVERS-521](https://jira.mongodb.org/browse/DRIVERS-521) (going back to 2018) but I don't think this question was answered. Is there any reason we needed to introduce a new scheme and couldn't control this via a driver option (akin to how we configure FLE and versioned API)? For instance, a driver option that toggles whether the driver prepends `_mongodb._tcp` and defaults to true for BC.

I only bring this up because there are a host of third-party libraries that track valid schemes for MongoDB connection strings. Those presently have `mongodb://` and `mongodb+srv://` hard-coded and I presume they would eventually need to be updated to accept `mongodb+rawsrv://` as well.

Reply:
A good question. Another option may be to add service name and protocol to URI options, either as new, named options or as key-value pairs in a field like `authMechanismProperties` (you can currently set `SERVICE_NAME` through `authMechanismProperties` for GSSAPI).

Reply:
I was originally thinking that we could let users specify anything for an SRV record, but the advantage of underscore prefixing is that it lets us tell where the "hostname.domainname" parts are.  On reflection, I would be supportive of keeping only `mongodb+srv` and adding a URI parameter to set the service name to something other than "_mongodb".  This is comparable to how we default the Kerberos service name to "mongodb" but let people set their own.